javascript/ql/src/semmle/javascript/frameworks/HTTP.qll

```diff
@@ -475,4 +476,118 @@ module HTTP {
   class CookieCryptographicKey extends CryptographicKey {
     CookieCryptographicKey() { this = any(CookieMiddlewareInstance instance).getASecretKey() }
   }
+
+  /**
+   * An object that contains one or more potential route handlers.
+   */
+  class RouteHandlerCandidateContainer extends DataFlow::Node {
+    RouteHandlerCandidateContainer::Range self;
+
+    RouteHandlerCandidateContainer() { this = self }
+
+    /**
+     * Gets the route handler in this container that is accessed at `access`.
+     */
+    DataFlow::SourceNode getRouteHandler(DataFlow::SourceNode access) {
```

Comment:
All implementations of this method return a `RouteHandlerCandidate`, so shouldn't the type reflect this? 
(Then you can remove the two `result instanceof RouteHandlerCandidate`)

Reply:
The intention was to hide the fact that `RouteHandlerCandidate` was used under the hood, but I guess the `RouteHandlerCandidateContainer` name is a spoiler in that regard :). I will leave it as is, and see if we end up changing the `RouteHandlerCandidateContainer` name instead due to some other discussion...